sdk/test-utils/recorder/README.md

```diff
@@ -1,9 +1,523 @@
+- How to avoid writing mocked versions of our HTTP API?
+- How to protect sensitive data from our live tests?
+- How to write tests that support parallelism?
+- How to write isomorphic tests for NodeJS and the Browsers?
 
-This library assumes that the tests rely on mocha test framework(and karma test runner for browser tests).
+Our NPM package `@azure/test-utils-recorder` attempts to provide an answer for
+those questions, as you'll be able to see throughout this README.
+
+This library provides interfaces and helper methods to equip the SDKs in the
+azure-sdk-for-js repo with the recording and playback capabilities for the
+tests, it targets HTTP requests in both Node.js and the Browsers.
+
+test-utils-recorder, as part of the Test Utils available in this repository, it
+is supposed to be added only as a devDependency and should be used only for the
+tests of an sdk.
+
+## Index
+
+- [Key concepts](#key-concepts).
+- [Getting started](#getting-started).
+    - [Installing the package](#installing-the-package).
+    - [Configuring your project](#configuring-your-project).
+- [Examples](#examples).
+  - [How to record](#how-to-record).
+  - [How to playback](#how-to-playback).
+  - [Updating existing recordings](#updating-existing-recordings).
+  - [Skipping tests](#skipping-tests).
+  - [Securing sensitive data](#securing-sensitive-data).
+  - [Ever-changing tests](#ever-changing-tests).
+  - [Supporting parallelism](#supporting-parallelism).
+  - [Isomorphic tests](#isomorphic-tests).
+- [Troubleshooting](#troubleshooting).
+- [Next steps](#next-steps).
+- [Contributing](#contributing).
+
+## Key concepts
+
+- To **record** means to intercept any HTTP request (specifically, every
+  http.request call and browser's XMLHttpRequest calls), store it in a file,
+  then store the response received from the live resource that was originally
+  targeted. We use a couple of mechanisms to do this for NodeJS and the
+  browser. The output files are stored in `recordings/node/*` and in
+  `recordings/browser/*`, which are relative to the root of the project you're
+  working on.
+- To **playback** means to intercept any HTTP request and to respond it with the
+  stored response of a previously recorded matching request.
+- **Sensitive information** means content that should not be shared publicly.
+  Content like passwords, unique identifiers or personal information should be
+  cleaned up from the recordings. Some functionality is provided to fix this
+  problem. You can read more at [Securing sensitive data](#securing-sensitive-data).
+
+## Getting started
+
+We're about to go through how to set up your project to use the
+`@azure/test-utils-recorder` package.  We'll be using git and
+[rush](https://rushjs.io). We understand that both tools can be challenging to
+use, but we expect you to know how to use them by the time you decide to use
+this package. If you encounter any problem with these tools, please let us know
```

Comment:
The wording of the sentence feels a bit scary.  We might want to say that "We'll be using git and rush in this readme. If you're not familiar with these tools, we'll provide ample samples to help you get started."

Reply:
Neither git or rush are simple things to tackle. Do you really want us to add documentation for both tools?

Reply:
No. What I'm saying is rather than scaring your audience, let them know if they're not familiar with the tools that you'll be providing examples of their use to help them get started. They need to read additional docs for the full story, of course.

Reply:
I did change this paragraph a bit though.

Reply:
What I'm trying to say, but perhaps not well, is don't call tools "challenging". This is called editorializing (ie giving your opinions on something). It's generally something to leave out of technical writing.

Also, don't require the users that "we expect you to know how to use them by the time you decide to use this package". This is a bit too strong. We will give them lots of examples they can copy paste if they're not familiar with, eg, Rust. That's okay. They don't have to be an experience Rush user to be able to use this library.

Instead, you could say that this tutorial assumes some familiarity with those tools and link them to the documentation.

Reply:
Pointing to the "rush" related section in the Readme that is present at the root of this repo might help.